PageRank/markov_chain.py

```diff
@@ -91,38 +106,150 @@ def __init__(
+                u_c = self.vertices[u].get_center()
+                vec = v_c - u_c
+                unit_vec = vec / np.linalg.norm(vec)
+
+                arrow_start = u_c + unit_vec * self.vertices[u].radius
+                arrow_end = v_c - unit_vec * self.vertices[v].radius
+                edge.put_start_and_end_on(arrow_start, arrow_end)
+
+        self.add_updater(update_edges)
+
+    def add_edge_buff(
+        self,
+        edge: tuple[Hashable, Hashable],
+        edge_type: type[Mobject] = None,
+        edge_config: dict = None,
+    ):
+        """
+        Custom function to add edges to our Markov Chain
+        """
+        if edge_config is None:
+            edge_config = self.default_edge_config.copy()
+        added_mobjects = []
+        for v in edge:
+            if v not in self.vertices:
+                added_mobjects.append(self._add_vertex(v))
         u, v = edge
-        arrow = self.edges[edge]
+
+        self._graph.add_edge(u, v)
+
+        base_edge_config = self.default_edge_config.copy()
+        base_edge_config.update(edge_config)
+        edge_config = base_edge_config
+        self._edge_config[(u, v)] = edge_config
+
         v_c = self.vertices[v].get_center()
         u_c = self.vertices[u].get_center()
         vec = v_c - u_c
         unit_vec = vec / np.linalg.norm(vec)
+
         arrow_start = u_c + unit_vec * self.vertices[u].radius
         arrow_end = v_c - unit_vec * self.vertices[v].radius
-        self.edges[edge] = Arrow(arrow_start, arrow_end)
+
+        edge_mobject = edge_type(
+            start=arrow_start, end=arrow_end, z_index=-100, **edge_config
+        )
+        self.edges[(u, v)] = edge_mobject
+
+        self.add(edge_mobject)
+        added_mobjects.append(edge_mobject)
+        return self.get_group_class()(*added_mobjects)
+
+    def add_markov_chain_edges(
+        self,
+        *edges: tuple[Hashable, Hashable],
+        edge_config: dict = None,
+        **kwargs,
+    ):
+        """
+        Custom function for our specific case of Markov Chains.
+        This function aims to make double arrows curved when two nodes
+        point to each other, leaving the other ones straight.
+
+        """
+
+        if edge_config is None:
+            edge_config = {}
+        non_edge_settings = {k: v for (k, v) in edge_config.items() if k not in edges}
+        base_edge_config = self.default_edge_config.copy()
+        base_edge_config.update(non_edge_settings)
+
+        base_edge_config = {e: base_edge_config.copy() for e in edges}
+        for e in edges:
+            base_edge_config[e].update(edge_config.get(e, {}))
+        edge_config = base_edge_config
+
+        edge_vertices = set(it.chain(*edges))
+        new_vertices = [v for v in edge_vertices if v not in self.vertices]
+        added_vertices = self.add_vertices(*new_vertices, **kwargs)
+
+        edge_types_dict = {}
+        for e in edges:
+            if (e[1], e[0]) in edges:
```

Comment:
I think the easiest way to add the flexibility that I mentioned in the comment is to check if the flag is enabled here, and only then add a curved arrow.